[depends] bump libjpeg-turbo to 2.0.4

Rechi opened the pull request:
## How Has This Been Tested?
compiled libjpeg-turbo and it's dependencies for Android (aarch64, arm, i686), iOS (aarch64, arm), macOS and tvOS

## Types of change
- [ ] **Bug fix** (non-breaking change which fixes an issue)
- [ ] **Clean up** (non-breaking change which removes non-working, unmaintained functionality)
- [x] **Improvement** (non-breaking change which improves existing functionality)
- [ ] **New feature** (non-breaking change which adds functionality)
- [ ] **Breaking change** (fix or feature that will cause existing functionality to change)
- [ ] **Cosmetic change** (non-breaking change that doesn't touch code)
- [ ] **None of the above** (please explain below)

## Checklist:
- [x] My code follows the **[Code Guidelines](https://github.com/xbmc/xbmc/blob/master/docs/CODE_GUIDELINES.md)** of this project 
- [ ] My change requires a change to the documentation, either Doxygen or wiki
- [ ] I have updated the documentation accordingly
- [x] I have read the **[Contributing](https://github.com/xbmc/xbmc/blob/master/docs/CONTRIBUTING.md)** document
- [ ] I have added tests to cover my change
- [x] All new and existing tests passed

AlwinEsch commented on tools/depends/target/Toolchain_binaddons.cmake.in:
```diff
@@ -1,4 +1,9 @@
 set(OS "@platform_os@")
+set(CMAKE_SYSTEM_PROCESSOR @host_cpu@)
```
This in case of Raspberry not correct enough, there it is only "arm" without something else, also on Android 32bit it is only "arm".

AlwinEsch replied:
See about Android here the standardized values of them https://github.com/Kitware/CMake/blob/master/Modules/Platform/Android-Determine.cmake#L243--L252 (on `..._PROC`)

AlwinEsch replied:
I'm torn, which would be better :smile:, if you go here it is fully OK https://cmake.org/cmake/help/latest/manual/cmake-toolchains.7.html.

Now looked over hundreds of CMakeLists.txt on the web, with the `set(CMAKE_SYSTEM_PROCESSOR ...)` many use it completely, but in the comparison I saw everywhere only for "arm" alone.

- The most make a `if (CMAKE_SYSTEM_PROCESSOR MATCHES "^arm")`
- Some are stupid and make a `if(CMAKE_SYSTEM_PROCESSOR STREQUAL "arm"`
- Some half stupid and make a `if(CMAKE_SYSTEM_PROCESSOR STREQUAL "arm" OR CMAKE_SYSTEM_PROCESSOR STREQUAL "armv7l")`

Then this path is actually OK, just a shame because of my work :cry: :laughing: :smile:.

Rechi committed:
[depends] bump libjpeg-turbo to 2.0.4

Rechi committed:
[toolchain] add CMAKE_SYSTEM_PROCESSOR

- enables clang-check, cppcheck and valgrind features for depends builds targeting the same system and processor as the host system
- libjpeg-turbo 2.0.0 switched to cmake and requires this variable

fuzzard commented on tools/depends/native/libjpeg-turbo/Makefile:
```diff
@@ -24,14 +19,14 @@ $(TARBALLS_LOCATION)/$(ARCHIVE):
 	cd $(PLATFORM); $(ARCHIVE_TOOL) $(ARCHIVE_TOOL_FLAGS) $(TARBALLS_LOCATION)/$(ARCHIVE)
-	cd $(PLATFORM); $(CONFIGURE)
+	cd $(PLATFORM); patch -p1 -i ../01-disable-executables.patch
+	cd $(PLATFORM); $(CMAKE) -B build -DCMAKE_TOOLCHAIN_FILE= -DCMAKE_ASM_NASM_COMPILER:FILEPATH=$(NATIVEPREFIX)/bin/yasm -DENABLE_SHARED:BOOL=OFF -DWITH_JPEG8:BOOL=ON
```
for the DCMAKE_ASM_NASM_COMPILER is it worth using the existing NASM var?

https://github.com/xbmc/xbmc/blob/353d61085acd8af33394051d207a5eabdd2fd657/tools/depends/native/config.site.native.in#L26

also, purely for reference if anyone looks in the future, libjpeg-turbo build docs say it can use either yasm or nasm, with some caveats around macos nasm min version apparently

> If using NASM, 2.10 or later is required.
If using NASM, 2.10 or later (except 2.11.08) is required for an x86-64 Mac build (2.11.08 does not work properly with libjpeg-turbo's x86-64 SIMD code when building macho64 objects.)
If using YASM, 1.2.0 or later is required.
If building on macOS, NASM or YASM can be obtained from MacPorts or Homebrew.
NOTE: Currently, if it is desirable to hide the SIMD function symbols in Mac executables or shared libraries that statically link with libjpeg-turbo, then NASM 2.14 or later or YASM must be used when building libjpeg-turbo.

https://github.com/libjpeg-turbo/libjpeg-turbo/blob/master/BUILDING.md

Rechi commented:
jenkins build this please
